app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -7,62 +7,43 @@ class EmitWebHookEvent < Jobs::Base
     RETRY_BACKOFF = 5
 
     def execute(args)
-      %i{
-        web_hook_id
-        event_type
-      }.each do |key|
-        raise Discourse::InvalidParameters.new(key) unless args[key].present?
-      end
-
-      @orig_args = args.dup
+      memoize_arguments(args)
+      validate_arguments!
 
-      web_hook = WebHook.find_by(id: args[:web_hook_id])
-      raise Discourse::InvalidParameters.new(:web_hook_id) if web_hook.blank?
-
-      unless ping_event?(args[:event_type])
-        return unless web_hook.active?
-
-        return if web_hook.group_ids.present? && (args[:group_id].present? ||
-          !web_hook.group_ids.include?(args[:group_id]))
+      unless ping_event?(event_type)
+        validate_argument!(:payload)
 
-        return if web_hook.category_ids.present? && (!args[:category_id].present? ||
-          !web_hook.category_ids.include?(args[:category_id]))
-
-        return if web_hook.tag_ids.present? && (args[:tag_ids].blank? ||
-          (web_hook.tag_ids & args[:tag_ids]).blank?)
-
-        raise Discourse::InvalidParameters.new(:payload) unless args[:payload].present?
-        args[:payload] = JSON.parse(args[:payload])
+        return if webhook_inactive?
+        return if group_webhook_invalid?
+        return if category_webhook_invalid?
+        return if tag_webhook_invalid?
       end
 
-      web_hook_request(args, web_hook)
+      send_webhook!
     end
 
     private
 
-    def guardian
-      Guardian.new(Discourse.system_user)
-    end
+    def validate_arguments!
+      %i{
+        web_hook_id
+        event_type
+      }.each do |key|
+        raise Discourse::InvalidParameters.new(key) unless arguments[key].present?
+      end
 
-    def ping_event?(event_type)
-      PING_EVENT == event_type.to_s
+      raise Discourse::InvalidParameters.new(:web_hook_id) if web_hook.blank?
     end
 
-    def build_web_hook_body(args, web_hook)
-      body = {}
-      event_type = args[:event_type].to_s
-
-      if ping_event?(event_type)
-        body[:ping] = 'OK'
-      else
-        body[event_type] = args[:payload]
-      end
+    def validate_argument!(key)
+      raise Discourse::InvalidParameters.new(key) unless arguments[key].present?
+    end
 
-      new_body = Plugin::Filter.apply(:after_build_web_hook_body, self, body)
-      MultiJson.dump(new_body)
+    def memoize_arguments(args)
```

Comment:
What is the point of the duplication here? Are we modifying the arguments? If so that seems dangerous.

Reply:
Yes, the `.dup` is because there is one argument we are modifying which is the: `retry_count` ... Thinking more about it, we don't need to do that, and I can just allocate an instance attribute for it.